Handle "Enter" key in manual barcode

patrick-du opened the pull request:
[Notion ticket](https://www.notion.so/uwblueprintexecs/Enter-Barcode-Page-pressing-the-Enter-key-does-not-pass-input-6c7eacc467fe4d73915e497ae63189d2)

## Changes
- ~~Updated `FormField.tsx` with a new `onKeyPress` prop~~
- ~~Reason for prop is because `TextField`, the component used in `FormField.tsx`, doesn't seem to have this as a prop~~ 
- ~~Additionally, onChange does not handle `React.KeyboardEvent`~~
- ~~On "Enter" key press, directs the user to the next step of the flow~~
- **After CR suggestions**
    - Use `onSubmit` to handle "Enter" key
## Testing
- Type in a barcode manually and press the "Enter" key on your keyboard
- This should redirect you to the next barcode page with your input from the previous page in the barcode field 
- If there was no input, you will stay on the same page

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [x] fill out `Changes`
  - [ ] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [ ] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [x] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

patrick-du committed:
Update FormField props

petorsong commented:
`onKeyPress` is an action that is no longer recommended as it has been removed from web standards (which is why material UI may have chosen to not support it) - [see this doc for more info](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onkeypress)

instead, i would prefer to leverage an action like `onSubmit` instead so we're not explicitly listening for a specific key press (i.e. enter) and instead just submitting the form normally, as if we had clicked the submit button

Commit:
Remove FormField changes and handle form submit

petorsong reviewed (approved):
real straightforward 🥇

JasonYG reviewed (dismissed):
Nice and simple change :)

petorsong reviewed (changes requested):
see new comments on notion ticket for additional changes necessary
https://www.notion.so/uwblueprintexecs/Enter-Barcode-Page-pressing-the-Enter-key-does-not-pass-input-6c7eacc467fe4d73915e497ae63189d2

rickymao commented:
LGTM!

patrick-du committed:
Add check for empty input

patrick-du committed:
Ensure barcode is not empty for form submission

patrick-du commented:
For the additional changes, I added a disabled check for the button for the visual effect, and for users that press the "Enter" key, I added an additional check in the method that handles the button, `handleEnterBarcode`.

rickymao reviewed (dismissed):
Looks Spooky To Me 👻

petorsong commented on src/components/EnterBarcode/EnterBarcodePage.tsx:
```diff
@@ -21,31 +21,35 @@ const EnterBarcodePage = ({
-        // Redirect to patient profile
-        history.push(
-          `/events/${patientEventId}/ccps/${patientCcpId}/patients/${id}`
+    if (barcode !== '') {
```
this check doesn't seem necessary - if we only call `handleEnterBarcode` when the form is submitted (i.e. when `barcode` is not an empty string), then this condition is always true as we cannot submit the form when `barcode` is an empty string

patrick-du replied:
Sounds good - in the new commit, I performed the check within `onSubmit` rather than inside of `handleEnterBarcode` since this function should only be routing the user rather than worrying about empty strings

patrick-du replied:
Also, I prevented the reloading with `e.preventDefault()` - hopefully that's the correct way to do so

petorsong reviewed (changes requested):
i'm seeing some weird behaviour when i hit enter in the input field while it's empty - it basically reloads the page but appends `?` to the route; ideally we don't do anything when trying to hit enter with an empty barcode, but if this proves difficult we can keep this current behaviour (not ideal though)

patrick-du committed:
Make changes according to CR

patrick-du commented on src/components/EnterBarcode/EnterBarcodePage.tsx:
```diff
@@ -67,7 +67,11 @@ const EnterBarcodePage = ({
-        <form>
+        <form
+          onSubmit={(e) =>
+            barcode !== '' ? handleEnterBarcode() : e.preventDefault()
```
Is this alright or should it be put in a function and called onSubmit instead?
```
const handleFormSubmission = (e) => {
     barcode !== '' ? handleEnterBarcode() : e.preventDefault()
}

...

<form onSubmit={(e) => handleFormSubmission(e)}>
```

petorsong replied:
i think this is just a matter of preference, up to you if you want to make the change - it can be cleaner and easier to edit in the future

also small nit for your example - `<form onSubmit={handleFormSubmit}>` should work as well - if you are gonna make the change i would test it first though

patrick-du replied:
Ok I've updated it - this ticket should be complete 😅

petorsong reviewed (dismissed):
:shipit:

patrick-du committed:
Store logic in function

petorsong reviewed (approved):
🥇

JasonYG commented on src/components/EnterBarcode/EnterBarcodePage.tsx:
```diff
@@ -49,6 +49,10 @@ const EnterBarcodePage = ({
   };
 
+  const handleFormSubmit = (e) => {
+    return barcode !== '' ? handleEnterBarcode() : e.preventDefault();
```
just wanted to confirm, we have `e.preventDefault()` so the form doesn't submit if the barcode is empty, right?

JasonYG commented:
Changes look good!!

rickymao reviewed (approved):
Looks spooky to me!
